### PR TITLE
Move absolute path finder from open_mfdataset to own function

### PR DESCRIPTION
**Pull request by Illviljan**
A simple refactor to make it easier to retrieve the proper paths that `open_mfdataset` uses and passes on the engine.

I've been thinking how to make use of DataTree and one idea I wanted to try was:
* Open file (using` _find_absolute_path`).
* Get all groups in the file.
* For each group run `xr.open_mfdataset(..., group=group)`

### CODE REVIEW

File: xarray/backends/common.py [hunk 1]
```diff
@@ -37,6 +57,64 @@ def _normalize_path(path):
     return path
 
 
+def _find_absolute_paths(
+    paths: str | NestedSequence[str | os.PathLike], **kwargs
```
Comment (headtr1ck) [inline xarray/backends/common.py:61]:
```suggestion
    paths: str | os.PathLike | NestedSequence[str | os.PathLike], **kwargs
```

File: xarray/backends/common.py [hunk 2]
```diff
@@ -37,6 +57,64 @@ def _normalize_path(path):
...
+        elif is_remote_uri(paths):
+            raise ValueError(
+                "cannot do wild-card matching for paths that are remote URLs "
+                f"unless engine='zarr' is specified. Got paths: {paths}. "
+                "Instead, supply paths as an explicit list of strings."
+            )
+        else:
+            paths = sorted(glob(_normalize_path(paths)))
+    elif isinstance(paths, os.PathLike):
+        paths = [os.fspath(paths)]
```
Comment (headtr1ck) [inline xarray/backends/common.py:111]: I know that this is just moved from the previous location, but is there a reason why we don't convert to absolute path here and below?

**Review by headtr1ck — APPROVED** [review]

File: xarray/backends/common.py (hunk 1 above)
Comment (headtr1ck) [reply xarray/backends/common.py:61]: Actually, why did you type this as a nested sequence? If I understand the code correctly only a "normal" sequence should be supported (even though it won't crash, just not parse anything).

File: xarray/backends/common.py (hunk 2 above)
Comment (Illviljan) [reply xarray/backends/common.py:111]: I'm not sure, it's a little odd to me as well. 

I haven't really felt the need for os.PathLike in my normal usage so I don't have much experience with that part though.

**Comment by TomNicholas** [comment]
@Illviljan see also https://github.com/xarray-contrib/datatree/issues/51

**Comment by Illviljan** [comment]
@TomNicholas I've found the current open_datatree implementations too focused on zarr and netcdf. Which is fine, it's the most popular after all. 
But I want to make use of custom engines so I'll play around on my side for a while.

**Pull request merged**